src/Mvc/Extensions.ApiDescription.Server/src/build/Microsoft.Extensions.ApiDescription.Server.targets

```diff
@@ -0,0 +1,39 @@
+﻿<?xml version="1.0" encoding="utf-8" standalone="no"?>
+<Project>
+  <PropertyGroup>
+    <_OpenApiDocumentsCache>$(BaseIntermediateOutputPath)$(MSBuildProjectName).OpenApiFiles.cache</_OpenApiDocumentsCache>
+  </PropertyGroup>
+  <ItemGroup Condition=" '$(OpenApiGenerateDocuments)' == 'true' ">
+    <ProjectCapability Include="OpenApiGenerateDocuments" />
+  </ItemGroup>
+
+  <Target Name="OpenApiGetDocuments" Returns="@(_OpenApiProjectDocuments)">
+    <ReadLinesFromFile File="$(_OpenApiDocumentsCache)">
```

Comment:
Is this actually a cache? or is this just the output of generating the documents?

Reply:
### TL;DR
Is your question aimed at my naming (and life) choices? I don't see much consistency in how similar MSBuild properties are named and chose something that fit.

Make a suggestion if you feel strongly please.

### Details
This file is many things&hellip;

It's a cache in the sense seen in lots of MSBuild code: It's a way to avoid redoing an action (calling a task, executing a too, whatever). In particular, this file avoids reopening the app whenever a client project calls `OpenApiGetDocuments`.

Yes, as an implementation detail, this file is _one of_ the files written as a result of running the tool. It lists the document files.

----
The file is also like many used in MSBuild code in that it summarizes multiple inputs or outputs of a task or tool. For example, I could do the `Inputs=$(TargetPath)` &hellip; `Outputs=$(_OpenApiDocumentsCache)` batching trick to avoid running `dotnet-getdocument` even if there are multiple document files.

----
Finally, this file is the communication mechanism between the `GenerateOpenApiDocuments` and `OpenApiGetDocuments` targets.